dbt/parser/schemas.py

```diff
@@ -61,186 +62,386 @@ def as_kwarg(key, value):
+            column_info.append({
+                'name': column_name,
+                'description': column.get('description', '')
+            })
+            for test in column.get('tests', []):
+                test_type, test_args = cls._build_v2_test_args(
+                    test, column_name
+                )
+                node = cls.build_node(
+                    model_name, package_name, test_type, test_args, root_dir,
+                    path, root_project, all_projects, macros
+                )
+                yield 'test', node
+
+        for test in model.get('tests', []):
+            # table tests don't inject any extra values, model name is
+            # available via `model.name`
+            test_type, test_args = cls._build_v2_test_args(test, None)
+            node = cls.build_node(model_name, package_name, test_type,
+                                  test_args, root_dir, path, root_project,
+                                  all_projects, macros)
+            yield 'test', node
+
+        patch = ParsedNodePatch(
+            name=model_name,
+            path=path,
+            description=model.get('description', ''),
+            columns=column_info
+        )
+        yield 'patch', patch
 
     @classmethod
     def load_and_parse(cls, package_name, root_project, all_projects, root_dir,
                        relative_dirs, macros=None):
-        extension = "[!.#~]*.yml"
-
         if dbt.flags.STRICT_MODE:
             dbt.contracts.project.ProjectList(**all_projects)
-
-        file_matches = dbt.clients.system.find_matching(
-            root_dir,
-            relative_dirs,
-            extension)
-
-        result = []
-
-        for file_match in file_matches:
-            file_contents = dbt.clients.system.load_file_contents(
-                file_match.get('absolute_path'), strip=False)
-
-            original_file_path = os.path.join(file_match.get('searched_path'),
-                                              file_match.get('relative_path'))
-
-            parts = dbt.utils.split_path(file_match.get('relative_path', ''))
-            name, _ = os.path.splitext(parts[-1])
-
-            result.append({
-                'name': name,
-                'root_path': root_dir,
-                'resource_type': NodeType.Test,
-                'path': file_match.get('relative_path'),
-                'original_file_path': original_file_path,
-                'package_name': package_name,
-                'raw_yml': file_contents
-            })
-
-        return cls.parse_schema_tests(result, root_project, all_projects,
-                                      macros)
+        new_tests = {}  # test unique ID -> ParsedNode
+        node_patches = {}  # model name -> dict
+
+        iterator = cls.find_schema_yml(package_name, root_dir, relative_dirs)
+
+        for original_file_path, test_yml in iterator:
+            version = test_yml.get('version', 1)
+            if version == 1:
```

Comment:
would it be crazy to warn people if they have a v2-looking spec in a v1 schema.yml? Like if one of the top-level keys is `models` but no version is specified? I can totally imagine forgetting to add the version every time, and just want to make sure that the error isn't super cryptic

Reply:
yeah, agree. I can work on the docs for this